music_assistant/providers/alexa/__init__.py

```diff
@@ -0,0 +1,314 @@
+            return
+
+        device_object = self.AlexaDevice()
+        await device_object.createobject(player_id, self.login)
+        api = AlexaAPI(device_object, self.login)
+        await api.play()
+
+        player.state = PlayerState.PLAYING
+        self.mass.players.update(player_id)
+
+    async def cmd_pause(self, player_id: str) -> None:
+        """Send PAUSE command to given player."""
+        if not (player := self.mass.players.get(player_id, raise_unavailable=False)):
+            return
+
+        device_object = self.AlexaDevice()
+        await device_object.createobject(player_id, self.login)
+        api = AlexaAPI(device_object, self.login)
+        await api.pause()
+
+        player.state = PlayerState.PLAYING
+        self.mass.players.update(player_id)
+
+    async def cmd_volume_set(self, player_id: str, volume_level: int) -> None:
+        """Send VOLUME_SET command to given player."""
+        if not (player := self.mass.players.get(player_id, raise_unavailable=False)):
+            return
+
+        device_object = self.AlexaDevice()
+        await device_object.createobject(player_id, self.login)
+        api = AlexaAPI(device_object, self.login)
+        await api.set_volume(volume_level / 100)
+
+        player.volume_level = volume_level
+        self.mass.players.update(player_id)
+
+    async def cmd_volume_mute(self, player_id: str, muted: bool) -> None:
+        """Send VOLUME MUTE command to given player."""
+        if not (player := self.mass.players.get(player_id, raise_unavailable=False)):
+            return
+
+        device_object = self.AlexaDevice()
+        await device_object.createobject(player_id, self.login)
+        api = AlexaAPI(device_object, self.login)
+        await api.set_volume(0)
+
+        player.volume_level = 0
+        self.mass.players.update(player_id)
+
+    async def cmd_seek(self, player_id: str, position: int) -> None:
+        """Handle SEEK command for given queue.
+
+        - player_id: player_id of the player to handle the command.
+        - position: position in seconds to seek to in the current playing item.
+        """
+
+    async def play_media(
+        self,
+        player_id: str,
+        media: PlayerMedia,
+    ) -> None:
+        """Handle PLAY MEDIA on given player.
+
+        This is called by the Players controller to start playing a mediaitem on the given player.
+        The provider's own implementation should work out how to handle this request.
+
+            - player_id: player_id of the player to handle the command.
+            - media: Details of the item that needs to be played on the player.
+        """
+        if not (player := self.mass.players.get(player_id)):
+            return
+
+        async with aiohttp.ClientSession() as session:
+            try:
+                async with session.post(
+                    "http://localhost:3000/ma/push-url",
```

Comment:
So what is this ?

Reply:
OK, this pushes the URL to the proxy. One question I have about this however is if you require one proxy instance per player ? You cant have one single proxy for multiple players ?

Also, if it makes it easier we can add an alias to the queue flow stream.
For example:
http://192.168.1.1:8097/stream/playerid.flac

Would that help ?

It still requires setting up the reverse proxy with SSL but it saves one step.